### PR TITLE
Requests

### PR DESCRIPTION
**Pull request by aandis**
@rohitpaulk have a look.

### CODE REVIEW

File: app/models/request.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+class Request < ActiveRecord::Base
+  belongs_to :requestee, :class_name => "User"
+  belongs_to :requester, :class_name => "User"
```
Comment (rohitpaulk) [inline app/models/request.rb]: Sender/Receiver :disappointed:

File: app/models/request.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  validates_presence_of :requestee
+  validates_presence_of :requester
+  validates_presence_of :item_type
+  validates_inclusion_of :item_type, in: ["Movie", "AndroidApp"]
+  validates_presence_of :status
+  validates_inclusion_of :status, in: ["pending", "sent", "seen", "successful"]
+
+  validates_uniqueness_of :item_type, scope: [:requestee, :requester],
```
Comment (rohitpaulk) [inline app/models/request.rb]: Why only `pending` and `sent`?

File: app/models/request.rb (hunk 1 above)
Comment (aandis) [inline app/models/request.rb]: Just following conventions you started man! :wink:

File: app/models/request.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    conditions: -> { where( status: ["pending","sent"] ) }
+
+  class RequestValidator < ActiveModel::Validator
```
Comment (rohitpaulk) [inline app/models/request.rb]: I think a class is overkill here, wouldn't using a custom method be simpler?

File: app/models/request.rb (hunk 2 above)
Comment (aandis) [inline app/models/request.rb]: If a request is `pending` or `sent` but not `seen` it implies the `requestee` hasn't yet acknowledged the request. So the other guy shouldn't be allowed to make more requests till then. Will probably need to check if the request is too old as well. If so, it should be allowed.

File: db/schema.rb [hunk 4]
```diff
@@ -19,6 +19,8 @@
     t.string   "icon_url"
     t.datetime "created_at",   null: false
     t.datetime "updated_at",   null: false
+    t.string   "rating"
```
Comment (rohitpaulk) [inline db/schema.rb]: I don't see a corresponding migration, why is this here?

File: spec/models/request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Request, :type => :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (rohitpaulk) [inline spec/models/request_spec.rb]: :tiger:

File: app/models/request.rb (hunk 3 above)
Comment (aandis) [inline app/models/request.rb]: It would later turn out to have it's own properties different than `Recommendation`. We shouldn't mix up things imo. Besides, we need to track request history as well. Read slack.

File: spec/models/request_spec.rb (hunk 5 above)
Comment (aandis) [inline spec/models/request_spec.rb]: No no no. :stuck_out_tongue:

File: app/models/request.rb (hunk 1 above)
Comment (rohitpaulk) [inline app/models/request.rb]: Don't make me feel guilty now

File: app/models/request.rb [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    self.status ||= 'pending'
+  end
+
+  def send_notification
+    requestee.send_notification(self.serializable_hash(:include => ["requester", "requestee"]))
+    self.status = 'sent'
+    save!
+  end
+
+  private
```
Comment (rohitpaulk) [inline app/models/request.rb]: :+1:

File: db/schema.rb (hunk 4 above)
Comment (aandis) [inline db/schema.rb]: It got added in 494ce10f980394e62b7a26e44f24d3ec7424fd1c. Must be a mixup while working on the apps branch. Will fix.

File: app/models/request.rb (hunk 6 above)
Comment (aandis) [inline app/models/request.rb]: ?

File: db/schema.rb (hunk 4 above)
Comment (aandis) [inline db/schema.rb]: Is there a way to amend that one commit to fix this and keep the child commits same?

File: app/models/request.rb (hunk 6 above)
Comment (rohitpaulk) [inline app/models/request.rb]: I like the privacy

File: db/schema.rb (hunk 4 above)
Comment (rohitpaulk) [inline db/schema.rb]: No easy way that I know of, no... What you could do is branch off from the commit behind the one you want to change, and cherry pick the commits one by one, amending them as needed. In our repo, I wouldn't really bother :P

Comment (aandis) [inline db/schema.rb]: Amended the latest commit. :sleeping:

File: app/models/request.rb (hunk 6 above)
Comment (aandis) [inline app/models/request.rb]: :feelsgood:
